examples/serverless/package.json

```
@@ -0,0 +1,32 @@
+{
+  "name": "serverless",
```

Oh maybe this is part of the issue! This can't be named `serverless` otherwise it'll conflict with the `serverless` we're installing in `devDependencies`. I think maybe doing the following will at least make it clearer why we're getting lint errors.

First, rebase `main` onto this branch so it's up to date, then check out the `package-lock` from there:

```
git pull --rebase origin main && git checkout main -- package-lock.json
```

Then change this package name to:

```suggestion
  "name": "@dotcom-reliability-kit/example-serverless",
```

Then run `npm install` which will update the package-lock to properly link between modules so there's no conflict with the `serverless` name.

---

Still getting url errors in the type checking though, which is odd. I'm trying to work out what's changed between the branches and I'm confused by this:

On the `main` branch, `url` was already installed via n-health:

```
npm ls --depth 999 url

dotcom-reliability-kit@0.0.0 /Users/rowan/Code/financial-times/dotcom-reliability-kit
└─┬ @dotcom-reliability-kit/example-express@0.0.0 -> ./examples/express
  └─┬ @financial-times/n-express@26.3.5
    └─┬ n-health@8.0.1
      └─┬ aws-sdk@2.1223.0
        └── url@0.10.3
```

```
npm ls --depth 999 n-health

dotcom-reliability-kit@0.0.0 /Users/rowan/Code/financial-times/dotcom-reliability-kit
└─┬ @dotcom-reliability-kit/example-express@0.0.0 -> ./examples/express
  └─┬ @financial-times/n-express@26.3.5
    └── n-health@8.0.1
```


But on the `CPREL-395-add-lambda-example` branch it is no longer picked up under `n-health`, it's only under `aws-sdk`:

```
npm ls --depth 999 url

dotcom-reliability-kit@0.0.0 /Users/rowan/Code/financial-times/dotcom-reliability-kit
└─┬ @dotcom-reliability-kit/example-serverless@0.0.0 -> ./examples/serverless
  └─┬ serverless@3.25.1
    └─┬ aws-sdk@2.1259.0
      └── url@0.10.3
```

```
npm ls --depth 999 n-health

dotcom-reliability-kit@0.0.0 /Users/rowan/Code/financial-times/dotcom-reliability-kit
└─┬ @dotcom-reliability-kit/example-express@0.0.0 -> ./examples/express
  └─┬ @financial-times/n-express@26.3.5
    └── n-health@8.0.1
```

(this is all after applying the changes suggested above)

---

I have had this before and think I maybe solved with a `// @ts-ignore` comment or by adding `@types/<module>` to the dev dependencies, but neither will work here as we're never actually importing `url` and `@types/url` does not exist.

I don't understand why yet, but [this answer](https://stackoverflow.com/a/54277766) solved it for me – no more errors relating to the `url` package and it still reports errors in our own modules. Maybe someone with more TypeScript knowledge can help here?

---

Doh thanks! That makes sense.

---

I have absolutely no idea what's going on with `types: []` but it works for me too 😢

---

@rowanmanning I think this is passing now 😬